File: bindings/nostr-sdk-ffi/bindings-python/examples/nwc.py
import asyncio

from nostr_sdk import init_logger, LogLevel, NostrWalletConnectUri, Nwc


async def main():
    # Init logger
    init_logger(LogLevel.INFO)

    # Parse NWC uri
    uri = NostrWalletConnectUri.parse("nostr+walletconnect://..")

    # Initialize NWC client
    nwc = Nwc(uri)

    info = await nwc.get_info()
    print(info)

    balance = await nwc.get_balance()
    print(f"Balance: {balance} SAT")


if __name__ == '__main__':
    asyncio.run(main())

File: bindings/nostr-sdk-js/examples/nwc.js
const { NWC, NostrWalletConnectURI, loadWasmAsync, initLogger, LogLevel } = require("../");

async function main() {
    await loadWasmAsync();

    initLogger(LogLevel.info());

    // Parse NWC uri
    let uri = NostrWalletConnectURI.parse("nostr+walletconnect://..");

    // Initialize NWC client
    let nwc = await new NWC(uri);

    // Get info
    let info = await nwc.getInfo();
    console.log("Supported methods: ", info.methods);

    // Get balance
    let balance = await nwc.getBalance();
    console.log("Balance: " + balance + " SAT");

    // Drop client
    nwc.free();
}

main();